### PR TITLE
Fix broken references in `docs/web_advanced.rst`

### PR DESCRIPTION
**Pull request by l1storez**
## What do these changes do?

 This change corrects multiple unrendered intersphinx class reference in the `web_advanced.rst` document.

## Are there changes in behavior for the user?

 No

## Related issue number

 #5518 

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by l1storez** [comment]
> /home/alexe/aiohttp/docs/web_advanced.rst:751: WARNING: py:meth reference target not found: UrlDispatcher.register_resource
>/home/alexe/aiohttp/docs/web_advanced.rst:851: WARNING: py:data reference target not found: zmq.SUB
>/home/alexe/aiohttp/docs/web_advanced.rst:851: WARNING: py:obj reference target not found: application['websockets']
>/home/alexe/aiohttp/docs/web_advanced.rst:896: WARNING: py:func reference target not found: listen_to_redis
>/home/alexe/aiohttp/docs/web_advanced.rst:971: WARNING: py:func reference target not found: aiohttp_debugtoolbar.setup

`UrlDispatcher.register_resource` could be `aiohttp.web.UrlDispatcher.add_resource` or `aiohttp.web.UrlDispatcher.named_resources`?
`listen_to_redis` is `readline.redisplay`?

**Comment by webknjaz** [comment]
> WARNING: py:data reference target not found: zmq.SUB

This could probably be fixed by adding https://pyzmq.rtfd.io/en/latest to the intersphinx mapping in `conf.py`

**Comment by webknjaz** [comment]
> `UrlDispatcher.register_resource` could be `aiohttp.web.UrlDispatcher.add_resource` or `aiohttp.web.UrlDispatcher.named_resources`?

No, the method exists, it's just not documented: https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_urldispatcher.py#L1023

File: docs/web_advanced.rst [hunk 1]
```diff
@@ -436,7 +436,7 @@ the keyword-only ``middlewares`` parameter::
 
 Internally, a single :ref:`request handler <aiohttp-web-handler>` is constructed
 by applying the middleware chain to the original handler in reverse order,
-and is called by the :class:`RequestHandler` as a regular *handler*.
+and is called by the :class:`aiohttp.web.RequestHandler` as a regular *handler*.
```
Comment (webknjaz) [inline docs/web_advanced.rst]: Don't forget to keep how it's being rendered, when possible:
```suggestion
and is called by the :class:`~aiohttp.web.RequestHandler` as a regular *handler*.
```

Comment (l1storez) [reply docs/web_advanced.rst]: forgot what tilde does:)

**Comment by webknjaz** [comment]
> `listen_to_redis` is `readline.redisplay`?

Definitely not. It's a reference to the inline code example here: https://github.com/l1storez/aiohttp/blob/5518-fix-web-adv/docs/web_advanced.rst#L866.

**Comment by webknjaz** [comment]
> > /home/alexe/aiohttp/docs/web_advanced.rst:971: WARNING: py:func reference target not found: aiohttp_debugtoolbar.setup

The project in question does not have a sphinx-based docs site to link against. So it's safe to ignore it too.

**Comment by l1storez** [comment]
Do you mean this https://github.com/l1storez/aiohttp/blame/5518-fix-web-adv/docs/web_advanced.rst#L866  ? It's the undocumented status?

**Comment by webknjaz** [comment]
Just ignore that one.

**Comment by l1storez** [comment]
Where is the problem?

File: docs/conf.py [hunk 2]
```diff
@@ -67,6 +67,7 @@
     "aiohttpsession": ("https://aiohttp-session.readthedocs.io/en/stable/", None),
     "aiohttpdemos": ("https://aiohttp-demos.readthedocs.io/en/latest/", None),
     "asynctest": ("https://asynctest.readthedocs.io/en/latest/", None),
+    "zmq.SUB": ("https://pyzmq.rtfd.io/en/latest/", None),
```
Comment (webknjaz) [inline docs/conf.py]:
```suggestion
    "pyzmq": ("https://pyzmq.rtfd.io/en/latest/", None),
```

File: docs/web_advanced.rst (hunk 1 above)
Comment (webknjaz) [reply docs/web_advanced.rst]: It influences the rendering

File: docs/web_advanced.rst [hunk 3]
```diff
@@ -849,9 +849,9 @@ sources (e.g. ZeroMQ, Redis Pub/Sub, AMQP, etc.) to react to received messages
 within the application.
 
 For example the background task could listen to ZeroMQ on
-:data:`zmq.SUB` socket, process and forward retrieved messages to
+``zmq.SUB`` socket, process and forward retrieved messages to
```
Comment (webknjaz) [inline docs/web_advanced.rst:852]: This completely removes any references effectively making that intersphinx mapping useless. Don't do this.
Instead, we need to figure out what this actually meant to reference. My bet it's "subscribe"

**Comment by webknjaz** [comment]
You forgot to fill out the PR template.

File: docs/web_advanced.rst (hunk 3 above)
Comment (l1storez) [reply docs/web_advanced.rst:852]: Don't understand the last sentence :confused:

**Comment by l1storez** [comment]
fill PR template

**Review by webknjaz — APPROVED** [review]

**Pull request merged**